public/components/create_report/create_report.tsx

```diff
@@ -0,0 +1,309 @@
+  { value: 'Wednesday', text: 'Wednesday' },
+  { value: 'Thursday', text: 'Thursday' },
+  { value: 'Friday', text: 'Friday' },
+  { value: 'Saturday', text: 'Saturday' },
+  { value: 'Sunday', text: 'Sunday' },
+];
+
+export function CreateReport() {
+  const [reportSettingsDashboard, setReportSettingsDashboard] = useState('');
+  const [deliveryEmailSubject, setDeliveryEmailSubject] = useState('');
+  const [deliveryEmailBody, setDeliveryEmailBody] = useState('');
+  const [scheduleRadioFutureDateSelected, setScheduleRadioFutureDateSelected] = useState(false);
+  const [scheduleRadioRecurringSelected, setScheduleRadioRecurringSelected] = useState(false);
+  const [scheduleUTCOffset, setScheduleUTCOffset] = useState(0);
+  const [scheduleFutureDate, setScheduleFutureDate] = useState(moment());
+  const [scheduleRecurringFrequency, setScheduleRecurringFrequency] = useState('Daily');
+  const [scheduleRecurringUTCOffset, setScheduleRecurringUTCOffset] = useState(0);
+  const [scheduleRecurringWeeklyDayOfWeek, setScheduleRecurringWeeklyDayOfWeek] = useState('Monday');
+  const [scheduleRecurringStartDate, setScheduleRecurringStartDate] = useState(moment());
+  const [scheduleRadioIdSelected, setScheduleRadioIdSelected] = useState(`${idPrefix}7`);
+
+  const onChangeReportSettingsDashboard = (e) => {
+    setReportSettingsDashboard(e.target.value);
+  }
+
+  const onChangeDeliveryEmailSubject = (e) => {
+    setDeliveryEmailSubject(e.target.value);
+  }
+
+  const onChangeDeliveryEmailBody = (e) => {;
+    setDeliveryEmailBody(e.target.value);
+  }
+
+  const onChangeScheduleSettingsRadio = optionId => {
+    setScheduleRadioIdSelected(optionId);
+    if (optionId === `${idPrefix}7`) {
+      setScheduleRadioFutureDateSelected(false);
+      setScheduleRadioRecurringSelected(false);
+    }
+    else if (optionId === `${idPrefix}8`) {
+      setScheduleRadioFutureDateSelected(true);
+      setScheduleRadioRecurringSelected(false);    
+    }
+    else if (optionId === `${idPrefix}9`) {
+      setScheduleRadioFutureDateSelected(false);
+      setScheduleRadioRecurringSelected(true);
+    }
+  };
+
+  const ScheduleRadio = () => {
+    const radios = [
+      {
+        id: `${idPrefix}7`,
+        label: 'Now',
+      },
+      {
+        id: `${idPrefix}8`,
+        label: 'Future Date',
+      },
+      {
+        id: `${idPrefix}9`,
+        label: 'Recurring',
+      },
+    ];
+    return (
+      <EuiRadioGroup
+        options={radios}
+        idSelected={scheduleRadioIdSelected}
+        onChange={onChangeScheduleSettingsRadio}
+        name="scheduleRadioGroup"
+      />
+    );
+  };
+  
+  const timezone_options = [
+    { value: -4, text: 'EDT -04:00' },
+    { value: -5, text: 'CDT -05:00' },
+    { value: -6, text: 'MDT -06:00' },
+    { value: -7, text: 'MST/PDT -07:00' },
+    { value: -8, text: 'AKDT -08:00' },
+    { value: -10, text: 'HST -10:00' }
+  ];
+
+  const ScheduleFutureDatePicker = () => {  
```

Comment:
You could move all individual components out to simplify and better readability.